app/models/bin.rb

```diff
@@ -1,4 +1,8 @@
 class Bin < ActiveRecord::Base
   validates :payload, presence: true
+  before_create :generate_random_id
 
+  def generate_random_id
+    self.random_id = SecureRandom.hex(30)
```

Comment:
I would suggest to use the `generate_unique_secure_token(length: MINIMUM_TOKEN_LENGTH)` method of ActiveRecord: https://api.rubyonrails.org/classes/ActiveRecord/SecureToken/ClassMethods.html

 https://github.com/rails/rails/blob/main/activerecord/lib/active_record/secure_token.rb